_data-prepper/pipelines/configuration/processors/list-to-map.md

```diff
@@ -0,0 +1,303 @@
+---
+layout: default
+title: List to map processor
+parent: Processors
+grand_parent: Pipelines
+nav_order: 55
+---
+
+# list_to_map
+
+The `list-to-map` processor converts a list of objects from an event where each object contains a `key` field, into a map of target keys.
+
+## Configuration
+
+The following table describes the configuration options to generate target keys to object mappings:
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`key` | Yes | String | The key of the fields to be extracted as keys in the generated mappings
+`source` | Yes | String | The list of objects with `key` fields that be converted to keys based on the generated mappings.
```

Comment:
This might be better since the `key` fields will be used as keys in the generated map. They are not dependent or based on the map.
```suggestion
`source` | Yes | String | The list of objects with `key` fields that will be converted to keys in the generated mappings.
```